agent/consul/rpc.go

```diff
@@ -379,6 +374,15 @@ RUN_QUERY:
 		if err := s.consistentRead(); err != nil {
 			return err
 		}
+
+		if isReady, waitCh := s.IsReadyForConsistentReads(); !isReady {
```

Comment:
Here's what I'm thinking - let's roll this the logic here inside of `consistentRead()` so we don't make the RPC function any more complex (the interaction with the timer is going to be ugly and the special case seems worth supporting. 

Instead of using the channel, let's add a simple atomic value to the server. This can be initialized to zero in `revokeLeadership()` and set to one in `establishLeadership()` I realized that `establishLeadership()` is already set up to be the place where post-barrier actions get called, and it will only get called once.

Inside of `consistentRead()` we can just poll this to see if it turns to 1. I realized that there's precedent for this already in rpc.go https://github.com/hashicorp/consul/blob/master/agent/consul/rpc.go#L222-L232 using the `RPCHoldTimeout`, which is designed to let RPCs hang for a little bit waiting for the leader to be available. This lets us centralize this additional check into the same `consistentRead()` fn that everybody should be calling, and keeps it from interacting with the other timer, which is really for blocking queries. It'll give us the same behavior of waiting out this period for a little while so we don't feed these back to the caller.